core/container/pool/pool.odin

```diff
@@ -0,0 +1,96 @@
+package container_pool
+
+import "core:fmt"
+
+// Single bit to mark a slot as active
+PoolFlag :: bit_set[0..<64;u64]
+
+// Dynamically resizable generic data pool
+Pool :: struct($type: typeid) {
+    data : [dynamic]type,
+    flag : [dynamic]PoolFlag,
+    free : [dynamic]int
+}
+
+// Reserve memory for the pool
+init :: proc($type: typeid, data_cnt : uint, free_cnt : uint = 8) -> Pool(type) {
+    pool : Pool(type)
+    reserve(&pool.data, data_cnt)
+    reserve(&pool.flag, (data_cnt / 64) + 1)
+    reserve(&pool.free, free_cnt)
+    return pool
+}
+
+is_index_valid :: proc(pool : ^Pool($T), index : int) -> (is_valid : bool) {
+    if index >= 0 && index < len(pool.data) {
+        flag := pool.flag[index / 64]
+        
+        is_valid = index % 64 in flag
+    }
+    return
+}
+
+active_count :: proc(pool : ^Pool($T)) -> int {
+    total : int
+    for f in pool.flag do total += card(f)
+    return total
+}
+
+get_ref :: proc(pool : ^Pool($T), #any_int index : int, loc := #caller_location ) -> (value : ^T = nil, ok : bool) #optional_ok {
+    if is_index_valid(pool, index) {
+        value = &pool.data[index]
+        ok = true
+    } else {
+        fmt.eprintfln("%v: Index is out of bounds", loc)
```

Comment:
This function should either:

- Panic if an illegal state has been reached
- Return `ok = false`, if the operation could not be performed

You are printing to the error stream *AND* returning `false`. This does not really make any sense. If you provide an index that is out of bounds, it is clearly a bug in your usage code. You should not provide an index that is out of bounds when accessing the pool. The best behaviour for a program if it detects a bug is to panic (Which is to crash, but here the error is simply ignored).

It does not make sense to return `ok = false` here either by the same reasoning. A bug should not be returned as a boolean false.